core/src/ledger_cleanup_service.rs

```diff
@@ -59,43 +71,121 @@ impl LedgerCleanupService {
         Self { t_cleanup }
     }
 
+    fn find_slots_to_clean(
+        blockstore: &Arc<Blockstore>,
+        root: Slot,
+        max_ledger_shreds: u64,
+    ) -> (u64, Slot, Slot) {
+        let mut shreds = Vec::new();
+        let mut iterate_time = Measure::start("iterate_time");
+        let mut total_shreds = 0;
+        let mut first_slot = 0;
+        for (i, (slot, meta)) in blockstore.slot_meta_iterator(0).unwrap().enumerate() {
+            if i == 0 {
+                first_slot = slot;
+                debug!("purge: searching from slot: {}", slot);
+            }
+            // Not exact since non-full slots will have holes
+            total_shreds += meta.received;
```

Comment:
Is there a limit on `meta.received`?

Reply:
https://github.com/solana-labs/solana/blob/master/core/src/shred_fetch_stage.rs#L38, 32,000 shreds is the max it seems

Reply:
At ~8000 slots/hr, somebody with 10% of the stake fills up about 10% of the `DEFAULT_MAX_LEDGER_SHREDS` every hour, doesn't seem too bad...